File: fixtures/scale/php/medium/src/Ast/FunctionCall.php
<?php

namespace Calc\Ast;

class FunctionCall implements Node
{
    /** @param Node[] $arguments */
    public function __construct(public string $name, public array $arguments)
    {
    }

    public function describe(): string
    {
        $parts = array_map(fn (Node $n) => $n->describe(), $this->arguments);
        return $this->name . '(' . implode(', ', $parts) . ')';
    }
}

File: fixtures/scale/csharp/medium/Ast/FunctionCall.cs
using System.Collections.Generic;

namespace Calc.Ast
{
    public class FunctionCall : Node
    {
        public string Name { get; }
        public List<Node> Arguments { get; }

        public FunctionCall(string name, List<Node> arguments)
        {
            Name = name;
            Arguments = arguments;
        }

        public override string Describe()
        {
            var parts = Arguments.ConvertAll(a => a.Describe());
            return Name + "(" + string.Join(", ", parts) + ")";
        }
    }
}
